### PR TITLE
feat(license_preview): Created an endpoint to fetch all license shortnames

### PR DESCRIPTION
**Pull request by deo002**
## Changes

-  Create an endpoint to fetch shortnames of all licenses. This data will be displayed as options in the multiselect input which adds licenses to a obligation.

## Submitter Checklist

- [ ] Includes tests (if there is a feature changed/added)
- [X] Includes docs ( if changes are user facing)
- [X] I have tested my changes locally.

### CODE REVIEW

File: pkg/api/licenses.go [hunk 1]
```diff
@@ -1152,3 +1152,60 @@ func ExportLicenses(c *gin.Context) {
 	c.Header("Content-Disposition", fmt.Sprintf("attachment; filename=%s", fileName))
 	c.JSON(http.StatusOK, &licenses)
 }
+
+// GetAllLicensePreviews retrieves a list of topics and types of all obligations
```
Comment (avinal) [inline pkg/api/licenses.go:1156]: Is the comment correct? We are only fetching Licenses, not obligations.

File: pkg/api/licenses.go [hunk 2]
```diff
@@ -1152,3 +1152,60 @@ func ExportLicenses(c *gin.Context) {
...
+//	@Summary		Get shortnames of all active licenses
+//	@Description	Get shortnames of all active licenses from the service
+//	@Id				GetAllLicensePreviews
+//	@Tags			Licenses
+//	@Accept			json
+//	@Produce		json
+//	@Param			active	query		bool	true	"Active license only"
+//	@Success		200		{object}	models.LicensePreviewResponse
+//	@Failure		400		{object}	models.LicenseError	"Invalid active value"
+//	@Failure		500		{object}	models.LicenseError	"Unable to fetch obligations"
```
Comment (avinal) [inline pkg/api/licenses.go:1167]: See above

**Review by avinal — CHANGES_REQUESTED** [review]
Just an idea, would it be feasible to develop an endpoint that returns fields on request?

File: pkg/api/licenses.go (hunk 1 above)
Comment (deo002) [reply pkg/api/licenses.go:1156]: Yes, my bad

**Comment by deo002** [comment]
> Just an idea, would it be feasible to develop an endpoint that returns fields on request?

Yep, nice idea. If we get more UI screens with different fields of the same data being displayed, the endpoint sure will save a lot of code and effort. 
But I guess this should suffice for now.
What do you think @GMishx ?

**Comment by GMishx** [comment]
> > Just an idea, would it be feasible to develop an endpoint that returns fields on request?
> 
> Yep, nice idea. If we get more UI screens with different fields of the same data being displayed, the endpoint sure will save a lot of code and effort. But I guess this should suffice for now. What do you think @GMishx ?

I too agree we can add that functionality if we are moving towards GraphQL or gRPC. For now, this should be good enough.

File: pkg/api/licenses.go [hunk 3]
```diff
@@ -1152,3 +1152,60 @@ func ExportLicenses(c *gin.Context) {
...
+			Message:   "Invalid active value",
+			Error:     fmt.Sprintf("Parsing failed for value '%s'", active),
+			Path:      c.Request.URL.Path,
+			Timestamp: time.Now().Format(time.RFC3339),
+		}
+		c.JSON(http.StatusBadRequest, er)
+		return
+	}
+	query := db.DB.Model(&models.LicenseDB{})
+	query.Where("active = ?", parsedActive)
```
Comment (GMishx) [inline pkg/api/licenses.go:1189]: Field name is `rf_active`

https://github.com/fossology/LicenseDb/blob/7940f0a8f736b23396ec18509b5fb1c9380074f3/pkg/models/types.go#L34

**Review by GMishx — CHANGES_REQUESTED** [review]

**Review by GMishx — APPROVED** [review]

**Pull request merged**